util/hash.h

```diff
@@ -14,19 +14,35 @@
 #include <stdint.h>
 
 #include "rocksdb/slice.h"
+#include "util/murmurhash.h"
 
 namespace rocksdb {
 
+// Non-persistent hash. Only used for in-memory data structure
+// The hash results are applicable to change.
+extern uint64_t NPHash64(const char* data, size_t n, uint32_t seed);
+
 extern uint32_t Hash(const char* data, size_t n, uint32_t seed);
 
 inline uint32_t BloomHash(const Slice& key) {
   return Hash(key.data(), key.size(), 0xbc9f1d34);
 }
 
+inline uint64_t GetSliceNPHash64(const Slice& s) {
+  return NPHash64(s.data(), s.size(), 0);
+}
+
 inline uint32_t GetSliceHash(const Slice& s) {
   return Hash(s.data(), s.size(), 397);
 }
 
+inline uint64_t NPHash64(const char* data, size_t n, uint32_t seed) {
+  // Similar to murmur hash, but reduce one multiplication
+  // out of two. The hash quality is not as good as murmurhash.
+  return MURMUR_HASH(data, static_cast<int>(n),
```

Comment:
I see that `MurmurHash` is used in other places. According to https://github.com/facebook/rocksdb/blob/master/util/murmurhash.h#L19, I wonder whether we should always one rather than the other?

Reply:
Our Hash() is actually faster than murmurhash, despite it is 32-bit one. Our hash() only does one multiplication, rather than two by murmurhash. So it's too early to switch everything. Also, I believe everything that is not persistent has been switched.